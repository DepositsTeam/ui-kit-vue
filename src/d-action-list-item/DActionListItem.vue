<template>
  <d-box class="ui-action-list">
    <d-box class="ui-action-list__body__wrapper">
      <slot name="body">
        <d-box class="ui-action-list__body">
          <d-box
            class="ui-action-list__header__left"
            display="flex"
            align-items="center"
            gap="10px"
          >
            <slot name="icon">
              <d-box
                v-if="logo"
                class="ui-action-list__logo"
                is="img"
                :src="logo"
                :alt="title"
              ></d-box>
            </slot>
            <d-box>
              <slot name="title">
                <d-text
                  my0
                  font-face="circularSTD"
                  data-test="title"
                  color="#000000"
                  font-size="15px"
                  margin-bottom="0"
                  >{{ title }}
                </d-text>
              </slot>
              <slot name="description">
                <d-text
                  v-if="description"
                  font-size="13px"
                  margin-top="5px"
                  my0
                  data-test="description"
                  font-face="circularSTD"
                  >{{ description }}
                </d-text>
              </slot>
            </d-box>
          </d-box>
          <d-box class="ui-action-list__header__right">
            <slot name="subtitle">
              <d-text
                class="ui-action-list__subtitle-text"
                font-face="circularSTD"
                data-test="subtitle"
                v-if="subtitle"
                >{{ subtitle }}
              </d-text>
            </slot>
            <slot name="action">
              <d-button
                v-if="buttonText"
                @click="emit('clicked')"
                :size="buttonSize"
                :smart-color="buttonSmartColor"
                :color-scheme="buttonColorScheme"
              >
                {{ buttonText }}
              </d-button>
            </slot>
          </d-box>
        </d-box>
      </slot>
    </d-box>
  </d-box>
</template>

<script setup>
import { DBox, DText, DButton } from "@/main";

const emit = defineEmits(["clicked"]);
defineProps({
  title: {
    type: String,
    required: true,
  },
  logo: {
    type: String,
  },
  description: {
    type: String,
  },
  subtitle: {
    type: String,
  },
  buttonText: {
    type: String,
  },
  buttonSize: {
    type: String,
    default: "large",
  },
  buttonColorScheme: {
    type: String,
    default: "default",
  },
  buttonSmartColor: {
    type: String,
  },
});
</script>

<style scoped lang="scss">
.ui-action-list {
  .ui-action-list__header__left {
    color: #8c97a7;
    font-size: 14px;
    font-style: normal;
    font-weight: 450;
    line-height: normal;
  }

  .ui-action-list__body__wrapper {
    .ui-action-list__body {
      display: flex;
      padding: 16px 24px;
      justify-content: space-between;
      align-items: center;
      align-self: stretch;
      border: 0.5px solid #e2e8f0;
      background: #fff;

      .ui-action-list__header__right {
        display: flex;
        align-items: center;
        gap: 16px;
        /**
        flex-direction: column;
         **/

        .ui-action-list__subtitle-text {
          color: #8c97a7;
          font-size: 14px;
          font-style: normal;
          font-weight: 450;
          line-height: 24px; /* 171.429% */
        }
      }

      .ui-action-list__logo {
        height: 32px;
        border-radius: 50%;
      }
    }
  }
}
</style>
