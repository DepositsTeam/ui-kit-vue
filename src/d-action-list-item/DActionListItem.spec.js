import { mount } from "@vue/test-utils";
import { expect, test } from "vitest";
import DActionListItem from "./DActionListItem.vue";

test("renders the component", async () => {
  expect(DActionListItem).to.not.be.undefined;

  const wrapper = mount(DActionListItem, {
    props: {
      title: "Testing stuff",
      subtitle: "Testing the random stuff out there",
      description: "Random description",
    },
  });

  expect(wrapper.get("[data-test='title']").text()).to.be.equal(
    "Testing stuff"
  );

  expect(wrapper.get("[data-test='subtitle']").text()).to.be.equal(
    "Testing the random stuff out there"
  );

  expect(wrapper.get("[data-test='description']").text()).to.be.equal(
    "Random description"
  );
});

test("matches snapshot", async () => {
  const wrapper = mount(DActionListItem);
  expect(wrapper.element).toMatchSnapshot();
});

test("renders the title", async () => {
  const wrapper = mount(DActionListItem, {
    props: {
      title: "Testing stuff",
      subtitle: "Testing the random stuff out there",
      description: "Random description",
      logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
    },
  });

  expect(wrapper.get("[data-test='title']").text()).to.be.equal(
    "Testing stuff"
  );

  expect(wrapper.get(".ui-action-list__logo").attributes("alt")).to.be.equal(
    "Testing stuff"
  );
});

test("properly renders slots", async () => {
  const wrapper = mount(DActionListItem, {
    props: {
      title: "Testing stuff",
      subtitle: "Testing the random stuff out there",
      description: "Random description",
      logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
    },
    slots: {
      action: "<button>Revoke Access</button>",
      subtitle: "<p id='subtitle-test'>Testing the random stuff out there</p>",
      description: "<h2>Random description</h2>",
      title: "<h1>Testing stuff</h1>",
      icon: "<img src='https://assets.ondeposits.com/img/logo/deposits/svg/logo_main.svg' />",
    },
  });

  expect(wrapper.find("button").exists()).to.be.true;
  expect(wrapper.find("button").text()).to.be.equal("Revoke Access");
  expect(wrapper.find("#subtitle-test").text()).to.be.equal(
    "Testing the random stuff out there"
  );
  expect(wrapper.find("h2").text()).to.be.equal("Random description");
  expect(wrapper.find("h1").text()).to.be.equal("Testing stuff");
  expect(wrapper.find("img").attributes("src")).to.be.equal(
    "https://assets.ondeposits.com/img/logo/deposits/svg/logo_main.svg"
  );
});

test("Emits event on click", async () => {
  const wrapper = mount(DActionListItem, {
    props: {
      title: "Testing stuff",
      subtitle: "Testing the random stuff out there",
      description: "Random description",
      logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
      buttonText: "Revoke access",
    },
  });

  await wrapper.find("button").trigger("click");

  const emittedEvent = wrapper.emitted("click");

  expect(emittedEvent).to.have.lengthOf(1);
});
