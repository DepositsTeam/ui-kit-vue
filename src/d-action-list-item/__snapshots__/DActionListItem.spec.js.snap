// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div
  class="circularSTD deposits-ui-box ui-action-list"
  data-ui-kit-auto_generated-cltvdguj3s=""
  data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
  data-v-d545dd22=""
  style="--logo-height: 32px;"
>
  
  <div
    class="circularSTD deposits-ui-box ui-action-list__body__wrapper"
    data-ui-kit-auto_generated-cltvdguj3s=""
    data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
    data-v-d545dd22=""
  >
    
    
    <div
      class="circularSTD deposits-ui-box ui-action-list__body"
      data-ui-kit-auto_generated-cltvdguj3s=""
      data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
      data-v-d545dd22=""
    >
      
      <div
        class="circularSTD deposits-ui-box ui-action-list__header__left"
        data-ui-kit-auto_generated-6d24ilyadn=""
        data-ui-kit-theming-rules-auto_generated-6d24ilyadn=""
        data-v-d545dd22=""
      >
        
        
        <!--v-if-->
        
        <div
          class="circularSTD deposits-ui-box"
          data-ui-kit-auto_generated-cltvdguj3s=""
          data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
          data-v-d545dd22=""
        >
          
          
          <p
            center="false"
            class="circularSTD deposits-ui-box ui-text my0 customFontSize"
            data-test="title"
            data-ui-kit-auto_generated-yoe3g8g6xy=""
            data-ui-kit-theming-rules-auto_generated-yoe3g8g6xy=""
            data-v-d545dd22=""
            equallineheight="false"
            mb0="false"
            ml0="false"
            mr0="false"
            mt0="false"
            mx0="false"
            my0="true"
            uppercase="false"
          >
            
            
            
            
            
          </p>
          
          
          <!--v-if-->
          
          
        </div>
        
      </div>
      <div
        class="circularSTD deposits-ui-box ui-action-list__header__right"
        data-ui-kit-auto_generated-cltvdguj3s=""
        data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
        data-v-d545dd22=""
      >
        
        
      </div>
      
    </div>
    
    
  </div>
  
</div>
`;
